cms/utils/title.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+from cms.models import Title
+
+
+def has_redirect_loop(title, language, redirect_path):
```

Comment:
I think the recursive check isn't necessary.
We only care about the title being modified, thus the check can be reduced to the something like:

```python
if redirect and redirect == self.cleaned_data['path']:
    raise ValidationError("You can't redirect to the same url.")
```

in the forms.py

There is still a condition where a loop can be created, redirect to a page whose title of same language redirects back to this one, but lets address that on a separate pull request with its own test.

Reply:
Thank you very much for the feedback! Why do you think a recursive check isn't necessary? It would resolve the addressed issue and even more complicated ones (A -> B -> C -> A).

Reply:
I want to tackle the first scenario (redirect to the same page) in this pr and the second scenario (redirect to a page which redirects to current one) in a separate pr, I would like to prevent redirecting to a page with a redirect set, this would avoid having to do the recursive check. I can't think of many use-cases where is needed to have such complex redirect tree.